onmt/modules/Decoder.lua

```diff
@@ -329,14 +330,25 @@ function Decoder:backward(batch, outputs, criterion)
                                                          { batch.size, batch.sourceLength, self.args.rnnSize })
 
   local loss = 0
+  local indvAvgLoss = torch.zeros(outputs[1]:size(1))
 
   for t = batch.targetLength, 1, -1 do
     -- Compute decoder output gradients.
     -- Note: This would typically be in the forward pass.
     local pred = self.generator:forward(outputs[t])
     local output = batch:getTargetOutput(t)
 
-    loss = loss + criterion:forward(pred, output)
+    if self.args.indvLoss then
+      local curLossSum = 0
+      for i=1, pred[1]:size(1) do
+        local tmpLoss = criterion:forward({pred[1][i]},{output[1][i]})
```

Comment:
This does not work with target features. For example with one target feature:

`pred` is in the form:

```
{
  1 : CudaTensor - size: 64x34058
  2 : CudaTensor - size: 64x9
}
```

and `output`:

```
{
  1 : CudaLongTensor - size: 64
  2 : CudaLongTensor - size: 64
}
```